### ISSUE TITLE
feat: wire cosine LR scheduler + validation early stopping in CudaTransformerTrainer

### ISSUE THREAD

**Issue by noahgift** [paiml/entrenar#275]
## Problem

CudaTransformerTrainer's pretrain loop uses constant LR after warmup — the `cosine` scheduler and `validation.patience` config fields are parsed but not implemented in the GPU training path.

## Impact

Run 8 diverged at step 220 because lr=2e-5 stayed constant when it should have decayed. Cosine annealing would have reduced LR from 2e-5 → ~0 over the remaining steps, preventing the gradient explosion.

### 1. Cosine LR Schedule
After warmup completes, apply:
```
lr(t) = lr_min + 0.5 * (lr_max - lr_min) * (1 + cos(π * (t - warmup) / (total - warmup)))
```

Currently in `cuda_trainer.rs`, the warmup logic exists but post-warmup LR is constant:
```rust
// Warmup: linear ramp
let lr = if step < warmup_steps {
    base_lr * (step as f32 / warmup_steps as f32)
} else {
    base_lr  // ← should be cosine decay
};
```

### 2. Validation Early Stopping
- Run forward-only pass on val split every `eval_every` steps
- Track val loss EMA
- Stop if val loss increases for `patience` consecutive evals
- Save best-val-loss checkpoint separately

### 3. Resume from Checkpoint
- Load LoRA weights from `.apr` checkpoint
- Restore optimizer state (Adam m/v buffers)
- Continue step counter from saved state

## References

- entrenar#274 (Run 8 divergence — root cause is missing scheduler)
- Config: `ssc-chat-qwen3-4b-qlora-v2.yaml` has `scheduler.name: cosine` but it's ignored
